File: 1]. DSA + CP/1]. DSA/1]. Data Structures/01]. Array/C++/_027)_Maximum_Subarray_Sum.cpp
#include<iostream>
using namespace std;

int maxSubarraySum(int arr[], int n)
{
    int res = arr[0];
    for(int i=0; i<n; i++)
    {
        int curr = 0;
        for(int j=i; j<n; j++)
        {
            curr += arr[j];
            res = max(res, curr);
        }
    }
    return res;
}

int main()
{
    int arr[] = {2, 3, -8, 7, -1, 2, 3}, n=7;
    cout<<"Maximum SubArray Sum = "<<maxSubarraySum(arr,n)<<endl;
}

File: 1]. DSA + CP/1]. DSA/1]. Data Structures/01]. Array/Dart/_027)_Maximum_Subarray_Sum.dart
import 'dart:math';

int maxSubArraySum(arr) {
  int res = arr[0];

  for (int i = 0; i < arr.length; i++) {
    int curr = 0;
    for (int j = i; j < arr.length; j++) {
      curr += arr[j] as int;
      res = max(res, curr);
    }
  }
  return res;
}

void main() {
  var arr = [2, 3, -8, 7, -1, 2, 3];
  print('Maximum sub-array sum is ${maxSubArraySum(arr)}.');
}
